jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdUserPrincipal.java

```diff
@@ -0,0 +1,50 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security.openid;
+
+import java.io.Serializable;
+import java.security.Principal;
+
+public class OpenIdUserPrincipal implements Principal, Serializable
```

Comment:
No point in making this class `Serializable` when its only field `_credentials` is not serializable.

Reply:
Removed the `Serializable`.

Reply:
So how is this auth going to work in a cluster?  If we have a non sticky load balancer, we need the authentication to move with the session - hence it needs to be serializable.

Reply:
I made `OpenIdUserPrincipal`, `OpenIdCredentials` and `OpenIdConfiguration` all `Serializable`. So now the `SessionAuthentication` used will allow the authentication to move with the session in a clustered environment.